scripting/lua/LuaCallWrapper.h

```diff
@@ -196,7 +197,7 @@ class LuaMethodWrapper <U, R(T:: *)(P1)const, method>
 		if(!S.tryGet(1,obj))
 			return S.retVoid();
 
-		P1 p1;
+		std::remove_cv_t<std::remove_reference_t<P1>> p1;
```

Comment:
Why is this needed?

Reply:
@kambala-decapitator In library series some types in VCMI library is converted from passing by value to passing by a const reference. If this change is not implemented, then Lua will breaks on const references passed in this templates. 

This change will strip constness and reference from such types, and just generate a new temporary value.

Reply:
To clarify - this is actual behavior that *is* happening and not yet another "improvement" from clang-tidy?

In that case perhaps better to add using at the very start of the class:
`using Parameter1 = td::remove_cv_t<std::remove_reference_t<P1>>;`
And then declare variables of type `Parameter1`

Reply:
@IvanSavenko This is not another improvement from clang-tidy. But it will noticed only if you pass const reference to this templates.

Will do your suggestion.

Reply:
@IvanSavenko your suggestion done for using.